ssl/statem/statem_lib.c

```diff
@@ -113,6 +113,11 @@ int tls_setup_handshake(SSL *s)
         }
         if (SSL_IS_FIRST_HANDSHAKE(s)) {
             s->ctx->stats.sess_accept++;
+        } else if ((s->options & SSL_OP_NO_RENEGOTIATION)) {
+            /* Renegotiation is disabled */
+            SSLerr(SSL_F_TLS_SETUP_HANDSHAKE, SSL_R_NO_RENEGOTIATION);
+            ssl3_send_alert(s, SSL3_AL_WARNING, SSL_AD_NO_RENEGOTIATION);
```

Comment:
Should we ever get here? We prevent the user calling SSL_renegotiate(), and we fail in the record layer if we receive a reneg ClientHello as a server, so I can't think how we would normally get to this point in the code. If this is a "should not happen" scenario then I'm wondering whether sending a warning no_renegotiation alert is the right thing to do (maybe a fatal internal_error?). Also perhaps a (soft) assert is worthwhile?